src/software/new_geom/linear_spline2d.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include <vector>
+
+#include "software/new_geom/point.h"
+#include "software/new_geom/polynomial2d.h"
+#include "software/new_geom/spline2d.h"
+
+/**
+ * A linear spline is a spline composed of a series of linear 2d polynomials, ie. a series
+ * of connected line segments
+ */
+class LinearSpline2d : Spline2d
+{
+   public:
+    LinearSpline2d() = delete;
+    /**
+     * Construct a spline by drawing line segments between consecutive
+     * Points
+     *
+     * @throws std::invalid_argument if points.size() == 0
+     *
+     * @param points Points on the spline
+     */
+    explicit LinearSpline2d(const std::vector<Point>& points);
+
+    /**
+     * Construct a spline by drawing line segments between consecutive
+     * Points
+     *
+     * @throws std::invalid_argument if points.size() == 0
+     *
+     * @param points Points on the spline
+     */
+    LinearSpline2d(const std::initializer_list<Point>& points);
+
+    /**
+     * Please see parent class for jdoc
+     */
+    const Point getValueAt(double val) const override;
+
+    /**
+     * Please see parent class for jdoc
+     */
+    size_t getNumKnots(void) const override;
+
+    /**
+     * Please see parent class for jdoc
+     */
+    const std::vector<Point> getKnots(void) const override;
+
+    /**
+     * Please see parent class for jdoc
+     */
+    const Point getStartPoint(void) const override;
+
+    /**
+     * Please see parent class for jdoc
+     */
+    const Point getEndPoint(void) const override;
+
+    /**
+     * Please see parent class for jdoc
+     */
+    const std::vector<SplineSegment2d> getSplineSegments() const override;
+
+   private:
+    // segments represent the polynomials that interpolate between
+    // time at index to index + 1
```

Comment:
Don't understand the "time at index to index + 1" part of this comment. Not sure what time or index you are referring to.

Reply:
Yeah, this is a bit of a rubbish comment, not sure where it comes from.... changed to:
```
    // The segments making up this spline, from the start to the end
```